### ISSUE TITLE
Double clicking on an edge causes a crash

### ISSUE THREAD

**Issue by shane-runsafe** [Vector35/binaryninja-api#1330]
The attached ARM binary has some odd disassembly that results in outbound edges to the middle of 32-bit thumb2 instructions.  Clicking on one of these outbound edges causes Binja to crash.

In this binary, the block at 0x0003cff0 has an outbound edge (in the True case) that goes to 0x3cf52 (which is in the middle of the valid instruction at 0x3cf50).  The GUI shows that edge going to an empty block, but double clicking on the edge causes the crash.

Here's some additional info.

>>> bv.get_basic_blocks_at(0x3cf52)
 [<block: thumb2@0x3cf14-0x3cf54>, <block: thumb2@0x3cf52-0x3cf54>]
 >>> block = bv.get_basic_blocks_at(0x3cf52)[1]
 >>> len(block)
 2
 >>> block.incoming_edges
 [<TrueBranch: thumb2@0x3cf52>]
 >>> block.outgoing_edges
 []
 >>> hex(block.start)
 '0x3cf52L'
 >>> block.instruction_count
 1
 >>> for instr in block:
 ...   print instr
 ... 
 >>> bv.get_basic_blocks_at(0x3cf52)[0].function
 <func: thumb2@0x3cc6c>
 >>> bv.get_basic_blocks_at(0x3cf52)[0].function == bv.get_basic_blocks_at(0x3cf52)[1].function 
 True

[linker.zip](https://github.com/Vector35/binaryninja-api/files/3094681/linker.zip)

**Comment by psifertex** [comment]
No idea what's /supposed/ to be there no matter how I look at that binary or what tool I use, but we certainly shouldn't be crashing. 

The fix for the crashing is easy. We'll definitely get that fixed. If you have any insight into the correct representation of what /should/ be there, let us know.

**Comment by shane-runsafe** [comment]
Your guess is as good as mine as to what is supposed to be there.  I didn't spend a lot of time looking at it, but the code around it seemed somewhat sane, so I'm not sure how that particular assembly got generated.

**Comment by psifertex** [comment]
No worries. I compared Ghidra, IDA, and BN and from what I can tell, IDA think it's pure data references there, Ghidra thinks it's code like us but likewise can't handle that particular branch (but screws up a previous jump table in that function). IDA gets the previous jump table but can't actually create a well-formed function, so it's all over the map. I figured if it was obvious what the correct issue with the disassembly was we could fix that too but since it was so off I wondered if it might not be something non-compiler generated, a modified binary or otherwise why it might be weird enough to throw off all three tools.
